legal-api/src/legal_api/services/filings/validations/correction.py

```diff
@@ -44,13 +49,14 @@ def validate(business: Business, filing: Dict) -> Error:
         path = '/filing/correction/correctedFilingId'
         msg.append({'error': _('Corrected filing is not a valid filing for this business.'), 'path': path})
 
-    if err := has_at_least_one_share_class(filing, 'incorporationApplication'):
-        msg.append({'error': _(err), 'path': '/filing/incorporationApplication/shareStructure'})
-
     # validations for firms
-    legal_type = filing.get('filing', {}).get('business', {}).get('legalType')
-    if legal_type and legal_type in [Business.LegalTypes.SOLE_PROP.value, Business.LegalTypes.PARTNERSHIP.value]:
-        _validate_firms_correction(business, filing, legal_type, msg)
+    if legal_type := filing.get('filing', {}).get('business', {}).get('legalType'):
+        if legal_type in [Business.LegalTypes.SOLE_PROP.value, Business.LegalTypes.PARTNERSHIP.value]:
+            _validate_firms_correction(business, filing, legal_type, msg)
+        elif legal_type in [Business.LegalTypes.COMP.value, Business.LegalTypes.BCOMP.value,
+                            Business.LegalTypes.LIMITED_CO.value, Business.LegalTypes.BC_ULC_COMPANY.value,
```

Comment:
I don't think `Business.LegalTypes.LIMITED_CO.value` is required?  think BC LTD fits under `Business.LegalTypes.COMP.value`